app/controllers/collections_controller.rb

```diff
@@ -131,12 +127,45 @@ def revoke_access
     redirect_to licences_path
   end
 
+  def web_create_collection
+    authorize! :web_create_collection, Collection
+    # Expose public licences and user created licences
+    licence_table = Licence.arel_table
+    @licences = Licence.where(licence_table[:private].eq(false).or(licence_table[:owner_id].eq(current_user.id)))
+    if request.post?
+      begin
+        # Check required fields
+        required_fields = {:collection_name => 'collection name', :collection_title => 'collection title'}
+        required_fields.each do |key, value|
+          raise ResponseError.new(400), "Required field '#{value}' is missing" if params[key].blank?
+        end
+
+        # Validate and sanitise additional metadata fields
+        additional_metadata = {}
```

Comment:
this and the below really belong in the `validate_additional_metadata` method, so the line looks like this:

`additional_metadata = validate_additional_metadata(params)`